src/edit/main.js

```diff
@@ -156,11 +156,22 @@ class ProseMirror {
       domDrop: new DOMSubscription
     }
 
-    if (opts.place && opts.place.appendChild)
+    if (opts.place && opts.place.appendChild) {
+      this.root = opts.place
+
+      // loop if root is not document (in light dom)
+      // only shadowRoot has property (in shadow dom)
+      while (this.root!== document && !this.root.host) {
```

Comment:
> If opts.place is a function provided by the consumer, it puts this.wrapper in the DOM.

Correct. We don't know when pm is mounted into DOM, it could be async. Or it's never in the DOM. Maybe this logic can't be just in the constructor?